// zudoku/lib/AuthenticationPlugin-DPCuR3xm.js
import { j as o } from "./jsx-runtime-C5mzlN2N.js";
import { LogOutIcon as a } from "lucide-react";
import { useEffect as i } from "react";
import { b as u, a as r } from "./chunk-BAXFHI7N-BVBOl9s0.js";
import { e as s } from "./hook-CldJlP5c.js";
const c = () => {
  const t = s(), [e] = u();
  return i(() => {
    var n;
    (n = t.authentication) == null || n.signIn({
      redirectTo: e.get("redirect") ?? void 0
    });
  }, [t.authentication, e]), null;
}, g = () => {
  const t = s(), e = r();
  return i(() => {
    var n;
    (n = t.authentication) == null || n.signOut().then(() => e("/"));
  }, []), null;
}, m = () => {
  const t = s();
  return i(() => {
    var e, n;
    ((e = t.authentication) == null ? void 0 : e.signUp()) ?? ((n = t.authentication) == null || n.signIn());
  }, [t.authentication]), null;
};
class d {
  getRoutes() {
    return [
      {
        path: "/signout",
        element: /* @__PURE__ */ o.jsx(g, {})
      },
      {
        path: "/signin",
        element: /* @__PURE__ */ o.jsx(c, {})
      },
      {
        path: "/signup",
        element: /* @__PURE__ */ o.jsx(m, {})
      }
    ];
  }
  getProfileMenuItems() {
    return [
      {
        label: "Logout",
        path: "/signout",
        category: "bottom",
        icon: a
      }
    ];
  }
}
export {
  d as A
};
